gcloud-java-core/src/main/java/com/google/gcloud/Service.java

```diff
@@ -16,6 +16,6 @@
 
 package com.google.gcloud;
 
-public interface Service<O extends ServiceOptions> {
-  O options();
+public interface Service<OptionsT extends ServiceOptions<?, OptionsT>> {
```

Comment:
That's a good question -- we definitely want to follow the open-source world's thoughts.

Seems the summary is:

Source|Suggestion
-----|------
Oracle|Use single characters (`E`, `K`, `V`)
Erwin Mueller (?)|Append `Type`
Google|Append `T`

(Source: http://stackoverflow.com/questions/2900881/generic-type-parameter-naming-convention-for-java-with-multiple-chars)